### PR TITLE
feat: initial configuration and setup with stubbed out docs pages

### PR DESCRIPTION
**Pull request by ryexley**
OK, this is the initial setup and implementation PR for [Docusaurus](https://v2.docusaurus.io/) as our engineering documentation setup. Most of this PR should just be Docusaurus scaffolding stuff that you won't need to look _too closely_ at, but it can give you an idea of how it's setup and organized.

There's a chicken-and-egg scenario where, this project won't build until we get the [`ui-common`](https://github.com/theartofeducation/ui-common) project setup to publish packages that can be consumed by dependent projects such as this one, because this project depends on some resources from that package, so we're going to need to get the PR for that project pushed through, and then get it updated with some CI jobs to handle publishing a package for it, and then we can update this project to pull it in as a dependency, and then it should build as we need it to.

But, for the sake of getting them both across the line, these initial PRs need to get merged in and in place so we can get that stuff going.

In the `/docs` folder is where I've put stubbed out files for the bulk of the engineering documentation that I think we want to get started on, but once this PR gets merged in, I think it would be appropriate for us to start creating cards for getting these individual documents fleshed out, with PRs for each of them individually.

Of course, feel free to hit me up with any questions you have about this.

### CODE REVIEW

File: docs/docusaurus-authoring-guide.mdx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+## Lists
+
+1. First ordered list item
+1. Another item
+   - Unordered sub-list.
+1. Actual numbers don't matter, just that it's a number
+   1. Ordered sub-list
+1. And another item.
+
+* Unordered list can use asterisks
```
Comment (theantichris) [inline docs/docusaurus-authoring-guide.mdx:50]: For our purposes, I'd like to pick one unorder list style and stick with it so all our docs are consistent.

File: docs/docusaurus-authoring-guide.mdx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+---
+
+## Line Breaks
+
+Here's a line for us to start with.
+
+This line is separated from the one above by two newlines, so it will be a _separate paragraph_.
+
+This line is also a separate paragraph, but... This line is only separated by a single newline, so it's a separate line in the _same paragraph_.
```
Comment (theantichris) [inline docs/docusaurus-authoring-guide.mdx:168]: Should probably add that there is a single newline because the line ending with 2 empty spaces.

**Review by theantichris — DISMISSED** [review]
Did you remove markdown linting?

**Comment by theantichris** [comment]
Branch rules need to be setup.

**Review by theantichris — CHANGES_REQUESTED** [review]
See comments.

File: docs/docusaurus-authoring-guide.mdx (hunk 1 above)
Comment (Thomas-Jean) [reply docs/docusaurus-authoring-guide.mdx:50]: I prefer * over - if we are picking

**Review by Thomas-Jean — DISMISSED** [review]

File: docs/docusaurus-authoring-guide.mdx (hunk 1 above)
Comment (ryexley) [reply docs/docusaurus-authoring-guide.mdx:50]: This document was one of the ones that gets installed with Docusaurus out of the box, so I honestly didn't even really look at it that closely, but I'm happy to make any changes that we want to make to it. 👍🏼

**Comment by ryexley** [comment]
I've updated branch protection rules on the repository.

**Comment by ryexley** [comment]
@theantichris I commented out markdown linting as part of the build process for now. I have a task to get MDX linting configured, since it's being used in the repository. Once I get markdown linting with MDX figured out, I will turn it back on, but I didn't want to let that hold up this PR, because it's been so long already. If you'd prefer, I can try to get that setup as part of this PR instead so we can get it in there. Let me know what you'd prefer.

File: docs/docusaurus-authoring-guide.mdx (hunk 2 above)
Comment (ryexley) [reply docs/docusaurus-authoring-guide.mdx:168]: Sorry, I'm not sure I understand what you're saying here.

**Review by Thomas-Jean — APPROVED** [review]

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -17,3 +17,9 @@ jobs:
 
       - name: Install
         run: yarn install
+
+      - name: Install markdownlinter-cli
+        run: npm install markdownlint-cli
```
Comment (theantichris) [inline .github/workflows/main.yml:22]: With this being a Node project we could probably remove this from the CI and have it as a project dependency like eslint. Thoughts?

**Review by theantichris — APPROVED** [review]

File: .github/workflows/main.yml (hunk 3 above)
Comment (ryexley) [reply .github/workflows/main.yml:22]: Yes, I am in the process of adding a card to Trello to do that very thing.

**Pull request merged**